### PR TITLE
Fix(#28847) CLI: push command no executing properly

### LINKED ISSUE #28847: CLI: push command no executing properly
### Parent Issue

_No response_

### Problem Statement

The command `dotcli push ./test-workspace --dry-run -e` is giving an error if is not executed inside of `./test-workspace`

```
[ERROR] ❗  Error in command [push] with message:
  Unable to access the path [./test-workspace/languages/test-workspace] check that it does exist and that you have read permissions on it.
java.io.IOException: Unable to access the path [./test-workspace/languages/test-workspace] check that it does exist and that you have read permissions on it.
	at com.dotcms.cli.command.language.LanguagePush.call(LanguagePush.java:118)
	at com.dotcms.cli.command.language.LanguagePush_Subclass.call$$superforward1(Unknown Source)
	at com.dotcms.cli.command.language.LanguagePush_Subclass$$function$$6.apply(Unknown Source)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.invoke(ActivateRequestContextInterceptor.java:124)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:33)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:33)
	at com.dotcms.cli.command.language.LanguagePush_Subclass.call(Unknown Source)
	at com.dotcms.cli.command.language.LanguagePush.call(LanguagePush.java:36)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at com.dotcms.cli.common.DotExecutionStrategy.execute(DotExecutionStrategy.java:87)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at com.dotcms.cli.command.PushCommand.call(PushCommand.java:83)
	at com.dotcms.cli.command.PushCommand_Subclass.call$$superforward1(Unknown Source)
	at com.dotcms.cli.command.PushCommand_Subclass$$function$$2.apply(Unknown Source)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.invoke(ActivateRequestContextInterceptor.java:129)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:33)
	at io.quarkus.arc.impl.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:33)
	at com.dotcms.cli.command.PushCommand_Subclass.call(Unknown Source)
	at com.dotcms.cli.command.PushCommand.call(PushCommand.java:22)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at com.dotcms.cli.common.DotExecutionStrategy.execute(DotExecutionStrategy.java:87)
	at io.quarkus.picocli.runtime.PicocliRunner$EventExecutionStrategy.execute(PicocliRunner.java:26)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at io.quarkus.picocli.runtime.PicocliRunner.run(PicocliRunner.java:40)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:129)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
```

Folder structure

```
|-- docker-compose.yml
|-- pull.sh
|-- push.sh
|-- start.sh
`-- test-workspace
    |-- content-types
    |-- files
    |-- languages
```

### Steps to Reproduce

- Using a local instance startup the local env using docker compose
- Run dotcli config and dotcli login if they haven't been run yet.
- if there is no data pulled into the current folder, run `dotcli pull --workspace ./test-workspace` to get the data from the local instace
- run `dotcli push ./test-workspace --dry-run -e` 

### Acceptance Criteria

id the command `dotcli push ./test-workspace --dry-run -e` it's have been executed the data in test-workspace should be pushed into the dotcms instance

### dotCMS Version

dotCLI 24.06.06
dotCMS 24.05.20

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by valentinogiardino**
### Proposed Changes
* Updated `PushMixin.path()` to ensure correct absolute path resolution.

### Screenshots
screen-recording-cli-push-command-relative-path

This PR fixes: https://github.com/dotCMS/core/issues/28847

This PR fixes: #28847

### CODE REVIEW

**Review by fabrizzio-dotCMS — CHANGES_REQUESTED** [review]
Two things: 
The path shown on the screenshots shows a "." in the middle which is incorrect 
you need to call normalize and then `toAbsolutePath()`

Also you need to write a test for this. 

That might work though I'm introducing some code to handle these cases in another PR

**Comment by valentinogiardino** [comment]
Output after applying `normalize()` as suggested by Fabrizzio:
screen-shot-cli-push-command

File: tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/PushCommandIT.java [hunk 1]
```diff
@@ -115,6 +115,39 @@ void testSimplePush() throws IOException {
         }
     }
 
+    /**
+     * This test checks for a simple push situation using a relative path where everything should work as expected.
```
Comment (fabrizzio-dotCMS) [inline tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/PushCommandIT.java:119]: can you please add a Given Scenario:  and then the scenario in which the test is supposed to happen
and then the Expected Result:  and the the result of the test
I know it is a bit painful but we're trying to be consistent across all our test documentation wise

**Review by fabrizzio-dotCMS — CHANGES_REQUESTED** [review]

File: tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/PushCommandIT.java (hunk 1 above)
Comment (valentinogiardino) [reply tools/dotcms-cli/cli/src/test/java/com/dotcms/cli/command/PushCommandIT.java:119]: Yes, for sure. No problem

**Review by fabrizzio-dotCMS — APPROVED** [review]

**Review by dcolina — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**